examples/nats-basic-secure-connection/publisher.bal

```diff
@@ -0,0 +1,27 @@
+import ballerinax/nats;
+
+public function main() returns error? {
+    // To secure the client connection using TLS/SSL, the client needs to be configured with
+    // a certificate file of the server.
+    // The [`nats:SecureSocket`](https://docs.central.ballerina.io/ballerinax/nats/latest/records/SecureSocket)
+    // record provides th   e SSL-related configurations of the client.
+    nats:SecureSocket sslConfig = { cert: "../resource/path/to/public.crt" };
+
+    // To secure the client connections using basic authentication, provide the credentials
+    // with the [`nats:Credentials`](https://docs.central.ballerina.io/ballerinax/nats/latest/records/Credentials) record.
+    nats:Credentials credentials = { username: "ballerina",
+                                     password: "ballerina@123" };
+
+    string message = "Hello from Ballerina";
+    // Initializes a NATS client with TLS/SSL and basic authentication.
+    nats:Client natsClient = check new(nats:DEFAULT_URL, auth = credentials,
```

Comment:
@aashikam Shall we use inline config declaration rather than external declaration and assignment. Something like https://ballerina.io/learn/by-example/http-service-mutual-ssl.html?is_ref_by_example=true IMO, it looks code nicer and clean. WDYT?

Reply:
Did it like this so the comment explaining the records can be put individually. Otherwise this would look like 
```ballerina
// Initializes a NATS client with TLS/SSL and basic authentication.
nats:Client natsClient = check new(nats:DEFAULT_URL, 

    // To secure the client connections using basic authentication, provide the credentials
    // with the [`nats:Credentials`](https://docs.central.ballerina.io/ballerinax/nats/latest/records/Credentials) record.
    auth = { 
             username: "ballerina",
             password: "ballerina@123" 
           },
           
    // To secure the client connection using TLS/SSL, the client needs to be configured with
    // a certificate file of the server.
    // The [`nats:SecureSocket`](https://docs.central.ballerina.io/ballerinax/nats/latest/records/SecureSocket)
    // record provides the SSL-related configurations of the client.
    secureSocket = {
            cert: "../resource/path/to/public.crt" 
           }
 );
```

Reply:
Yeah. When we get preview of the above it looks cleaner IMO. Let's use this for the moment and update the credentials also as stated with https://github.com/ballerina-platform/ballerina-distribution/pull/2055/files#r672045115